crates/ide/src/markdown_remove.rs

```diff
@@ -0,0 +1,16 @@
+use pulldown_cmark::{Event, Parser};
+
+pub fn remove_markdown(markdown: &str) -> String {
+    let mut out = String::new();
+    let parser = Parser::new(markdown);
+
+    for event in parser {
+        match event {
+            Event::Text(text) | Event::Code(text) => out.push_str(&text),
+            Event::SoftBreak | Event::HardBreak | Event::Rule => out.push('\n'),
+            _ => {}
```

Comment:
I think you should also push `* ` for list items. Otherwise

* foo
* bar

would be rendered as

```plain
foo
bar
```

which doesn't look like a list at all. You should also render tables using ascii art I think.

Reply:
I'm not sure where it goes, but is it possible to preserve the empty line in the example?

Reply:
@lnicola in the screenshot? that might be possible, and indeed look better.

@bjorn3 that would be a good idea, but i think this is enough when only the hover implementation uses it, as it only uses simple markdown. I can add a comment stating the lack off styling

Reply:
> `Event::End(Tag::CodeBlock(_))`

Was the empty line missing otherwise?

Looks good to me, (with the exception of the two boolean arguments function, which I'm not a fan of but is probably fine for now), but I didn't test it.